test/jdk/sun/net/www/protocol/https/HttpsURLConnection/B6216082.java

```diff
@@ -160,30 +198,30 @@ public boolean verify(String hostname, SSLSession session) {
     }
 }
 
-class SimpleHttpTransaction implements HttpCallback {
+class SimpleHttpTransaction implements HttpHandler {
     public boolean hasBadRequest = false;
 
     /*
      * Our http server which simply redirect first call
      */
-    public void request(HttpTransaction trans) {
+    public void handle(HttpExchange trans) {
         try {
             String path = trans.getRequestURI().getPath();
             if (path.equals("/")) {
                 // the first call, redirect it
                 String location = "/redirect";
-                trans.addResponseHeader("Location", location);
-                trans.sendResponse(302, "Moved Temporarily");
+                trans.getResponseHeaders().set("Location", location);
+                trans.sendResponseHeaders(302, -1);
             } else {
                 // if the bug exsits, it'll send 2 GET commands
                 // check 2nd GET here
-                String duplicatedGet = trans.getRequestHeader(null);
+                String duplicatedGet = trans.getRequestHeaders().getFirst(null);
```

Comment:
I'm not sure you will be testing the same thing here. You probably can't use the com.sun.net.httpserver.HttpServer for this.